### ISSUE TITLE
let_underscore_untyped get activated with clippy::pedantic while it should not

### ISSUE THREAD

**Issue by GilShoshan94** [rust-lang/rust-clippy#10731]
### Summary

Hi,

[`let_underscore_untyped`](https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_untyped) is categorized as `restriction` according to the documentation but get triggered with `clippy::pedantic`

Here is the diagnostic I get with the code sample I provided:
![image](https://user-images.githubusercontent.com/29143941/235491328-70c1da00-e7b9-4755-a8fd-84a19b7cd0ad.png)

It's written: "note: `#[warn(clippy::let_underscore_untyped)]` implied by `#[warn(clippy::pedantic)]`".

Please remove `let_underscore_untyped` from `pedantic` to fix this.

### Reproducer

I tried this code:

```rust
#![warn(clippy::pedantic)]

use tokio::sync::watch;

#[derive(Debug)]
pub struct Example {
    drop_switch: Option<watch::Sender<()>>,
    name:        String,
}

impl Example {
    #[must_use]
    pub fn new(name: String) -> Example {
        let (drop_switch, _drop_sig) = watch::channel(());
        let drop_switch = Some(drop_switch);

        Example { drop_switch, name }
    }

    pub fn hello(&self) {
        println!("Hello, my name is {}", self.name);
    }
}

impl Drop for Example {
    fn drop(&mut self) {
        if let Some(drop_switch) = self.drop_switch.take() {
            let _ = drop_switch.send(());
            drop(drop_switch);
        }
    }
}

#[tokio::main]
async fn main() {
    let name = "ABC".to_string();
    let a = Example::new(name);
    a.hello();
}
```

I expected to see this happen:

Instead, this happened:

### Version

```text
rustc 1.69.0 (84c898d65 2023-04-16)
binary: rustc
commit-hash: 84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc
commit-date: 2023-04-16
host: x86_64-pc-windows-msvc
release: 1.69.0
LLVM version: 15.0.7
```

**Comment by xFrednet** [comment]
The lint was originally added in the `pedantic` group. We only moved it to `restriction` recently. It looks like that change didn't make it into 1.69. In one or two releases, this should no longer happen :)

**Comment by GilShoshan94** [comment]
Thank you for the response.

I suggest to keep this issue open until the next release in order to validate that the change is indeed included.
Only then to close this issue.
